packages/react-accordion/src/components/AccordionPanel/useAccordionPanel.ts

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import { AccordionPanelProps, AccordionPanelState } from './AccordionPanel.types';
+import { useAccordionItemContext } from '../AccordionItem';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const accordionPanelShorthandProps = [];
+
+const mergeProps = makeMergeProps<AccordionPanelState>({ deepMerge: accordionPanelShorthandProps });
+
+/**
+ * Returns the props and state required to render the component
+ * @param props AccordionPanel properties
+ * @param ref reference to root HTMLElement of AccordionPanel
+ * @param defaultProps default values for the properties of AccordionPanel
+ */
+export const useAccordionPanel = (
+  props: AccordionPanelProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: AccordionPanelProps,
+): AccordionPanelState => {
+  const { panelId, headingId, open } = useAccordionItemContext();
+  const state = mergeProps(
+    {
+      ref: useMergedRefs(ref, React.useRef(null)),
+      id: props.id ?? panelId,
```

Comment:
```suggestion
      id: panelId,
```

nit: I suggest to double check, but it should be overridden by `resolveShorthandProps(props, accordionPanelShorthandProps)` in this chain.